azure-maven-plugin-lib/src/main/java/com/microsoft/azure/maven/artifacthandler/FTPArtifactHandlerImpl.java

```diff
@@ -38,28 +40,43 @@ protected String getDeploymentStageDirectory() {
 
     @Override
     public void publish(DeployTarget target) throws IOException, MojoExecutionException {
+        if (target.getApp() instanceof WebApp) {
+            prepareResources();
+        }
+        
+        assureStageDirectoryNotEmpty();
+
         final FTPUploader uploader = new FTPUploader(mojo.getLog());
         final PublishingProfile profile = target.getPublishingProfile();
         final String serverUrl = profile.ftpUrl().split("/", 2)[0];
 
-        if (this.mojo.getPluginName().equalsIgnoreCase(WEBAPP_PLUGIN_NAME)) {
-            final List<Resource> resources = this.mojo.getResources();
-            if (resources == null || resources.isEmpty()) {
-                mojo.getLog().info(NO_RESOURCES_CONFIG);
-                return;
-            }
-            Utils.copyResources(mojo.getProject(),
-                mojo.getSession(),
-                mojo.getMavenResourcesFiltering(),
-                resources,
-                getDeploymentStageDirectory());
-        }
-
         uploader.uploadDirectoryWithRetries(serverUrl,
             profile.ftpUsername(),
             profile.ftpPassword(),
             getDeploymentStageDirectory(),
             DEFAULT_WEBAPP_ROOT,
             DEFAULT_MAX_RETRY_TIMES);
+
+        if (target.getApp() instanceof FunctionApp) {
+            ((FunctionApp) target.getApp()).syncTriggers();
+        }
+    }
+
+    protected void prepareResources() throws IOException {
+        final List<Resource> resources = this.mojo.getResources();
+
+        if (resources != null && !resources.isEmpty()) {
+            Utils.copyResources(mojo.getProject(), mojo.getSession(),
+                mojo.getMavenResourcesFiltering(), resources, getDeploymentStageDirectory());
+        }
+    }
+
+    protected void assureStageDirectoryNotEmpty() throws MojoExecutionException {
+        final String stageDirectory = getDeploymentStageDirectory();
```

Comment:
The variable name is confusing here because of we both have `xxxDirectory` and `xxxFolder`

How about for the Those `String` typed variable, we call `xxxDirectoryPath`

For `File` typed variable, we call `xxxDirectory`